pkg/ingester/stream.go

```diff
@@ -288,30 +288,28 @@ func (s *stream) recordAndSendToTailers(record *wal.Record, entries []logproto.E
 	hasTailers := len(s.tailers) != 0
 	s.tailerMtx.RUnlock()
 	if hasTailers {
-		go func() {
```

Comment:
Hmm I don't understand the surrounding code, but from the `recordAndSendToTailers()` name and from the previous usage of a goroutine here, I assume this part should be as non-blocking as possible? If so, does it make sense to keep the goroutine here? We'd still handle the actual sending over a queue, but this part won't be blocking either.

Reply:
We are removing the goroutine and sending the stream to the queue. Since the queue is a buffered channel, it's a non-blocking opt (if chan is full, the stream is dropped)